core/Url.php

```diff
@@ -669,6 +664,24 @@ public static function isHostInUrls($host, $urls)
             }
         }
 
-        return in_array($host, self::$alwaysTrustedHosts);
+        return in_array($host, self::getAlwaysTrustedHosts());
+    }
+
+    /**
+     * List of hosts that are never checked for validity.
+     *
+     * @return array
+     */
+    private static function getAlwaysTrustedHosts()
+    {
+        return self::getLocalHostnames();
+    }
+
+    /**
+     * @return array
+     */
+    public static function getLocalHostnames()
+    {
+        return array('localhost', '127.0.0.1', '::1', '[::1]');
```

Comment:
I prefer not to use those because the IP and hostname would depend on the client issuing the request and  a visitor could be eg. from a different country or anywgere, which does not qualify as a "localhost" address....

but it's true that maybe this PR won't completely fix the issue reported in https://github.com/piwik/piwik/issues/8253 although it should fix #8272